contracts/strategies/StrategyAaveLendingPoolV2.sol

```diff
@@ -0,0 +1,172 @@
+import "./interfaces/aave/IAToken.sol";
+import "./interfaces/aave/IAaveIncentivesController.sol";
+import "./interfaces/aave/IStakeAave.sol";
+import "./interfaces/uniswap/IUniswapV2.sol";
+
+/**
+ * Deposits ERC20 token into Aave Lending Pool and issues stAaveLendingToken(e.g. stAaveLendingDAI) in L2. Holds aToken (Aave interest-bearing tokens).
+ */
+contract StrategyAaveLendingPoolV2 is IStrategy, Ownable {
+    using SafeERC20 for IERC20;
+    using Address for address;
+    using SafeMath for uint256;
+
+    // The address of Aave Lending Pool
+    address public lendingPool;
+
+    // Info of supplying erc20 token to Aave lending pool
+    // The symbol of the supplying token
+    string public symbol;
+    // The address of supplying token (e.g. DAI, USDT)
+    address public supplyToken;
+
+    // The address of Aave interest-bearing token (e.g. aDAI, aUSDT)
+    address public aToken;
+
+    address public controller;
+
+    // The address of Aave Incentives Controller
+    address public incentivesController;
+
+    // The address of Aave StakeToken contract
+    address public stakeToken;
+
+    // The address of Aave token
+    address public aave;
+
+    address public uniswap;
+    // The address of WETH token
+    address public weth;
+
+    constructor(
+        address _lendingPool,
+        string memory _symbol,
+        address _supplyToken,
+        address _aToken,
+        address _controller,
+        address _incentivesController,
+        address _stakeToken,
+        address _aave,
+        address _uniswap,
+        address _weth
+    ) {
+        lendingPool = _lendingPool;
+        symbol = _symbol;
+        supplyToken = _supplyToken;
+        aToken = _aToken;
+        controller = _controller;
+        incentivesController = _incentivesController;
+        stakeToken = _stakeToken;
+        aave = _aave;
+        uniswap = _uniswap;
+        weth = _weth;
+    }
+
+    function getAssetAddress() external view override returns (address) {
+        return supplyToken;
+    }
+
+    function syncBalance() external view override returns (uint256) {
+        // Supplying token(e.g. DAI, USDT) balance of this contract.
+        // aToken value is pegged to the value of supplying erc20 token at a 1:1 ratio.
+        uint256 supplyTokenBalance = IAToken(aToken).balanceOf(address(this));
+        return supplyTokenBalance;
+    }
+
+    function harvest() external override {
+
+        // 1. Claims the StkAAVE staking rewards
+        uint256 stakingRewards = IStakeAave(stakeToken).getTotalRewardsBalance(address(this));
+
+        if (stakingRewards > 0) {
+            IStakeAave(stakeToken).claimRewards(address(this), stakingRewards);
+        }
+
+        // 2. Redeems staked tokens, note that it won't be success while this account not in redeem-available period 
+        stakeToken.call(abi.encodeWithSignature("redeem(address,uint256)", address(this), IERC20(stakeToken).balanceOf(address(this))));
```

Comment:
Can we call their interface instead of using `encodeWithSignature`?

Reply:
Using _**call**_ here is to avoid revert even if the account is not in redeem-available period. No matter what the result is in this step, the step 3/4/5 should continue to be executed.

Reply:
Can we use `encodeWithSelector`?

https://github.com/celer-network/layer2-finance-contracts/blob/61ed0f17a15e8ba06778776ade1a82956a9de842/contracts/TransitionDisputer.sol#L240